### PR TITLE
[prometheus-elasticsearch-exporter] Introduce extraManifests value/option

### PR DESCRIPTION
**Pull request by ritmas**
#### What this PR does / why we need it

- This PR adds the `extraManifests` value/option
- It's possible to use `ES_USERNAME / ES_PASSWORD` as environment values for ES authentication, but it's not wise to hardcode them. This PR would allow to create extra manifest for getting them from any other Secret manager/vault and keep things clean, e.g. [External Secrets Operator](https://external-secrets.io/latest/api/externalsecret/)
- This kind of approach is already implemented in other charts in this repo

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-elasticsearch-exporter/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 description: Elasticsearch stats exporter for Prometheus
 name: prometheus-elasticsearch-exporter
-version: 6.0.0
+version: 6.0.1
```
Comment (zeritti) [inline charts/prometheus-elasticsearch-exporter/Chart.yaml:4]: A new feature as in this change needs raising the minor version number.

```suggestion
version: 6.1.0
```

**Review by zeritti — CHANGES_REQUESTED** [review]
Thank you, @ritmas, for your PR. Please, see my comments below.

File: charts/prometheus-elasticsearch-exporter/Chart.yaml (hunk 1 above)
Comment (ritmas) [reply charts/prometheus-elasticsearch-exporter/Chart.yaml]: @zeritti fixed it

**Comment by zeritti** [comment]
Thank you for the change. It's just occurred to me, will you add a test case in the `ci` directory for `extraManifests`? The example from the values file would be good enough. Thanks!

**Comment by ritmas** [comment]
@zeritti Added

File: charts/prometheus-elasticsearch-exporter/ci/extra-manifests.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+# Create extra manifests
+
+extraManifests:
+  - |
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+    labels:
+      name: prometheus-elasticsearch-exporter-extra
```
Comment (zeritti) [inline charts/prometheus-elasticsearch-exporter/ci/extra-manifests.yaml:10]:
```suggestion
    metadata:
      labels:
        name: prometheus-elasticsearch-exporter-extra
```

File: charts/prometheus-elasticsearch-exporter/values.yaml [hunk 3]
```diff
@@ -124,6 +124,17 @@ env: {}
...
 extraArgs: []
 
+# Extra manifests to deploy as an array
+extraManifests: []
+  # - |
+  #   apiVersion: v1
+  #   kind: ConfigMap
+  #   metadata:
+  #   labels:
+  #     name: prometheus-elasticsearch-exporter-extra
```
Comment (zeritti) [inline charts/prometheus-elasticsearch-exporter/values.yaml:134]:
```suggestion
  #   metadata:
  #     labels:
  #       name: prometheus-elasticsearch-exporter-extra
```

**Review by zeritti — CHANGES_REQUESTED** [review]

File: charts/prometheus-elasticsearch-exporter/ci/extra-manifests.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+---
+# Create extra manifests
+
+extraManifests:
+  - |
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      labels:
+        name: prometheus-elasticsearch-exporter-extra
```
Comment (zeritti) [inline charts/prometheus-elasticsearch-exporter/ci/extra-manifests.yaml:10]: Apologies, it is really too late. Name is missing, e.g.
```suggestion
    metadata:
      name: prometheus-elasticsearch-exporter-extra       
```

File: charts/prometheus-elasticsearch-exporter/values.yaml [hunk 5]
```diff
@@ -124,6 +124,17 @@ env: {}
...
 extraArgs: []
 
+# Extra manifests to deploy as an array
+extraManifests: []
+  # - |
+  #   apiVersion: v1
+  #   kind: ConfigMap
+  #   metadata:
+  #     labels:
+  #       name: prometheus-elasticsearch-exporter-extra
```
Comment (zeritti) [inline charts/prometheus-elasticsearch-exporter/values.yaml:134]:
```suggestion
  #   metadata:
  #     name: prometheus-elasticsearch-exporter-extra
  #       
```

**Review by zeritti — CHANGES_REQUESTED** [review]
Thank you, @ritmas. Please, rename the ci file to `extra-manifests-values.yaml`, it won't be used otherwise.

**Comment by ritmas** [comment]
@zeritti applied fixes

**Review by zeritti — APPROVED** [review]

**Pull request merged**